src/api/index.ts

```diff
@@ -96,14 +96,14 @@ function createExchangeApi(erc20Api: Erc20Api, injectedDependencies: DepositApiD
   return exchangeApi
 }
 
-function createTokenListApi(): TokenList {
-  const networks = [Network.Mainnet, Network.Rinkeby]
+function createTokenListApi(exchangeApi: ExchangeApi): TokenList {
+  const networkIds = [Network.Mainnet, Network.Rinkeby]
 
   let tokenListApi: TokenList
   if (process.env.MOCK_TOKEN_LIST === 'true') {
     tokenListApi = new TokenListApiMock(tokenList)
   } else {
-    tokenListApi = new TokenListApiImpl(networks)
+    tokenListApi = new TokenListApiImpl({ networkIds, exchangeApi })
```

Comment:
> mmm, can we avoid making an API depend on another API directly?

Yeah, that is a problem I thought about.

> maybe depend on a pure function? getTokenIdByAddress? 

That would still make it dependent on exchangeApi. The difference is that instead of depending on the whole api we depend on only 1 function.

> Otherwise, I think we created the service layer for this

I have an idea, not sure it'll work or whether it makes sense: Make `updateIdsPerNetwork` a service. 
For that we'd need:
1. expose a way to update the local token list (easy)
2. when tokens are updated, update subscribers (easy I guess?)
3. figure out a way to call `updateIdsPerNetwork` once per network, per page load, without circular dependencies (hard?)

`3` is hard because the current implementation invokes it when `getTokens` is called, and tracks it within the class. I think I might need to expose another service to encapsulate `getTokens` instead of allowing direct access to `tokenListApi`.
But then, how do I deal with the subscriptions?

Reply:
> That would still make it dependent on exchangeApi. The difference is that instead of depending on the whole api we depend on only 1 function.

Yes, this is not my prefered option. And yes, that's so true, however it would depend on a pure function that would be injected, so it will be blind to what someone injects there

>> Otherwise, I think we created the service layer for this
> I have an idea, not sure it'll work or whether it makes sense: Make updateIdsPerNetwork a service.

If I understand correctly, I like the proposal. By the look of `updateTokenIds` looks more like a service to me than an API. Coordinate other services. Actually, I don't think updateTokenIds should be a service, at least "public" I think it should be getTokens the service, right?

With that, you don't have circular dependencies, do you?

Reply:
Done what we discussed over slack.

Exposing 2 services for getting the tokens and for subscribing to updates.
TokenListApi should no longer be directly used as an api, but only from the service.
Except by other services. Don't want to create dependencies between services either.